pallets/accounts/src/lib.rs

```diff
@@ -446,16 +470,13 @@ pub mod pallet {
 
 					<EthReservedTickets<T>>::remove(&eth_key);
 					// also emit event
-					Self::deposit_event(TicketsMinted {
-						sender: sender.clone(),
-						balance: amount,
-					});
+					Self::deposit_event(TicketsMinted { sender: sender.clone(), balance: amount });
 				}
 			}
 			if <EthReservedNova<T>>::contains_key(&eth_key) {
 				let nova_amount = <EthReservedNova<T>>::get(&eth_key);
 				if let Some(amount) = nova_amount {
-					ensure!(!amount.is_zero(), Error::<T>::AccountAlreadyLinked);
+					ensure!(!amount.is_zero(), Error::<T>::SystematicFailure);
 					// Assign NOVA
 					<pallet_balances::Pallet<T> as Unbalanced<T::AccountId>>::set_balance(
```

Comment:
isn't set_balance dangerous? it "SETS" the balance to the number, shouldn't it be a mint call?

Reply:
I think you are looking at the the previous commit. I am ot using set balance anymore, i use Balanced deposit(). Mint is not possible with pallet balances, only with assets.

Reply:
https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1102
Are we sure?

Reply:
Cos deposit is more internel

Reply:
Done, used mint_into from fungible::Mutate.